src/link/index.tsx

```diff
@@ -14,6 +17,21 @@ const Link = React.forwardRef(
     const baseComponentProps = useBaseComponent('Link', {
       props: { color, external, fontSize, rel: props.rel, target: props.target, variant: props.variant },
     });
+
+    const analyticsMetadata: GeneratedAnalyticsMetadataLinkFragment = {
+      action: 'click',
+      detail: {
+        label: '',
```

Comment:
The more I look at these empty strings, the more I dislike them. Same suggestion as the previous PR: https://github.com/cloudscape-design/components/pull/2479#discussion_r1686203948

You will be grateful yourself for the fix, as it makes your future investigations of corner-cases easier

Reply:
What is your suggestion to address this? setting it to undefined will not work

Reply:
Why will this not work?

Reply:
because the lack of `label` indicates to the metadata resolution algorithm (https://github.com/cloudscape-design/component-toolkit/blob/main/src/internal/analytics-metadata/utils.ts#L11) that there is no label to resolve.

Reply:
There is an option to use selector :scope, but it would make the object more complicated

Reply:
So, there are 3 possible use-cases

1. `label` is `null` which prevents any resolution
1. `label` is a string which is treated as CSS-selector
1. If `label` is an empty string or no match found, resort to data attribute

I see a solution with extending LabelIdentifier interface https://github.com/cloudscape-design/component-toolkit/blob/main/src/internal/analytics-metadata/interfaces.ts#L37-L40

We can add `label: { disabled: true }` value instead of `null`  and then we will avoid null and empty string as different values

Reply:
Will go for `label: { root: "self" }`after merging https://github.com/cloudscape-design/component-toolkit/pull/84